sbin/health-checker.in

```diff
@@ -21,13 +21,29 @@ BTRFS_ID_DEFAULT=0
 SNAPSHOT_DEFAULT=""
 BTRFS_ID_CURRENT=0
 
+is_bls=0
+if [ -d /boot/efi/loader/entries ]; then
+    is_bls=1
+fi
+
 set_btrfs_id()
 {
     BTRFS_ID_DEFAULT=`btrfs subvolume get-default / | awk '{print $2}'`
     SNAPSHOT_DEFAULT="`btrfs subvolume get-default / | cut -d ' ' -f 9-`"
     BTRFS_ID_CURRENT=`findmnt --output OPTIONS --noheadings / | sed -e 's|.*subvolid=\([0-9]\+\).*|\1|g'`
 }
 
+get_snapshot()
+{
+    sed -e 's|.*@/\.snapshots/\([0-9]\+\)/snapshot.*|\1|g'
+}
+
+set_snapshot()
+{
+    SNAPSHOT_DEFAULT="$(btrfs subvolume get-default / | get_snapshot)"
+    SNAPSHOT_CURRENT="$(findmnt --output OPTIONS --noheadings / | get_snapshot)"
+}
```

Comment:
This function doesn't set the snapshot as the name implies - it's setting two variables containing snapshots.